src/components/molecules/CalendarBodyDates/CalendarBodyDates.tsx

```diff
@@ -0,0 +1,52 @@
+import { Calendar, GetDatePropsOptions } from 'dayzed'
+import { Box, Button, Text, Wrap, WrapItem } from '@chakra-ui/react'
+import { FC } from 'react'
+import { WEEKDAY_NAMES_SHORT } from '@/@types/Calendar'
+
+interface CalendarBodyDatesProps {
+  calendar: Calendar
+  getDateProps: (data: GetDatePropsOptions) => Record<string, any>
+}
+
+const CalendarBodyDates: FC<CalendarBodyDatesProps> = (props) => {
+  const { calendar, getDateProps } = props
+
+  return (
+    <Box>
+      <Wrap spacing={0}>
+        {WEEKDAY_NAMES_SHORT.map((weekday) => (
+          <WrapItem key={`${calendar.month}${calendar.year}${weekday}`} boxSize={`48px`} justifyContent={`center`}>
+            <Text textAlign={`center`} fontFamily={`body`} fontWeight={`medium`} margin={`auto`}>
+              {weekday[0]}
+            </Text>
+          </WrapItem>
+        ))}
+      </Wrap>
+      <Wrap spacing={0}>
+        {calendar.weeks.map((week, weekIndex) =>
+          week.map((dateObj, index) => {
+            const key = `${calendar.month}${calendar.year}${weekIndex}${index}`
+            if (!dateObj) {
+              return <WrapItem key={key} width={`calc(100% / 7)`}></WrapItem>
+            }
+            const { date, selected, selectable, today } = dateObj
+            let background = today ? `lightGray.200` : `transparent`
+            background = selected ? `lightGray.800` : background
+            background = !selectable ? `transparent` : background
+            return (
+              <WrapItem key={key} boxSize={`48px`} justifyContent={`center`}>
+                {selectable ? (
+                  <Button variant={`calendarDate`} color={selected ? `white` : `black`} backgroundColor={background} {...getDateProps({ dateObj })}>
+                    {date.getDate()}
+                  </Button>
+                ) : null}
+              </WrapItem>
```

Comment:
You can map over a list and genearte a list of react components outside of the return. Would be cleaner. Have some examples online.

Reply:
have extracted the generation of the React components out